### PR TITLE
port `FixedSizeCrop` from detection references to prototype transforms

### PR DESCRIPTION
**Pull request by pmeier**
```py
import torch
from torchvision.prototype import features, transforms


image = features.EncodedImage.from_path("test/assets/fakedata/logos/rgb_pytorch.png").decode()

bounding_boxes = features.BoundingBox(
    [[60, 30, 15, 15]],
    format=features.BoundingBoxFormat.CXCYWH,
    image_size=(100, 100),
    dtype=torch.float,
)

segmentation_masks = torch.zeros((1, 100, 100), dtype=torch.bool)
segmentation_masks[..., 24:36, 55:66] = True
segmentation_masks = features.SegmentationMask(segmentation_masks)
```

orig

```py
target = dict(
    boxes=bounding_boxes,
    masks=segmentation_masks,
)
sample = image, target

transform = transforms.FixedSizeCrop(size=(40, 40))

torch.manual_seed(0)

for i in range(5):
    transformed_image, transformed_target = transform(sample)
    transformed_bounding_boxes = transformed_target["boxes"]
    transformed_segmentation_masks = transformed_target["masks"]
```

    

```py
transform = transforms.FixedSizeCrop(size=(150, 150))
```

### CODE REVIEW

File: torchvision/prototype/transforms/_geometry.py [hunk 1]
```diff
@@ -611,3 +611,60 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
...
+        if params["needs_crop"]:
+            inpt = F.crop(
+                inpt,
+                top=params["top"],
+                left=params["left"],
+                height=params["height"],
+                width=params["width"],
+            )
+            # TODO: cull invalid bounding boxes and labels after we have resolved the preferred way in
+            #  https://github.com/pytorch/vision/pull/6401
```
Comment (datumbox) [inline torchvision/prototype/transforms/_geometry.py:665]: Are we blocked with this? Why can't we use the original approach of:
```python
is_valid = (boxes[:, 0] < boxes[:, 2]) & (boxes[:, 1] < boxes[:, 3])
```

and filtering anything that is not valid in the `inpt`?

Comment (vfdev-5) [reply torchvision/prototype/transforms/_geometry.py:665]: @pmeier is anything unresolved about that in https://github.com/pytorch/vision/pull/6401, can you detail ?

Comment (pmeier) [reply torchvision/prototype/transforms/_geometry.py]: Sorry, maybe the comment was a little misleading. This functionality relies on `query_bboxes` and `clamp_bounding_box` from #6401. Since there is no approval yet, there might be changes that directly affect my PR as well.

I've added 105bccd325d67509f2267f2fbeb617549968d516 to showcase how the solution will look like if #6401 is merged with no changes to the functions I listed above.

File: test/test_prototype_transforms.py [hunk 2]
```diff
@@ -1209,3 +1209,128 @@ def test__transform(self, mocker):
         transform(inpt_sentinel)
 
         mock.assert_called_once_with(inpt_sentinel, size=size_sentinel, interpolation=interpolation_sentinel)
+
+
+class TestFixedSizeCrop:
+    def test__get_params(self):
+        pass
```
Comment (pmeier) [inline test/test_prototype_transforms.py:1216]: TODO

**Review by datumbox — DISMISSED** [review]
LGTM, thanks!

File: torchvision/prototype/transforms/_geometry.py [hunk 3]
```diff
@@ -672,3 +672,87 @@ def _get_params(self, sample: Any) -> Dict[str, Any]:
...
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+        if params["needs_crop"]:
+            inpt = F.crop(
+                inpt,
+                top=params["top"],
+                left=params["left"],
+                height=params["height"],
+                width=params["width"],
+            )
+            if isinstance(inpt, (features.Label, features.SegmentationMask)):
```
Comment (datumbox) [inline torchvision/prototype/transforms/_geometry.py:747]: Why do we handle masks here like that?

@vfdev-5 do you need this on your PR?

Comment (vfdev-5) [reply torchvision/prototype/transforms/_geometry.py:747]: Yes, we need to apply that on one-hot encoded masks. Added that in my PR

File: torchvision/prototype/transforms/_geometry.py [hunk 4]
```diff
@@ -744,7 +744,7 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
                 height=params["height"],
                 width=params["width"],
             )
-            if isinstance(inpt, (features.Label, features.SegmentationMask)):
+            if isinstance(inpt, (features.Label, features.OneHotLabel, features.SegmentationMask)):
```
Comment (vfdev-5) [inline torchvision/prototype/transforms/_geometry.py:747]: Here you make an assumption that seg mask is one-hot encoded. In segmenation task, the mask is not one-hot encoded and thus this may give an unexpected behaviour.

**Review by datumbox — APPROVED** [review]
LGTM, thanks!

File: torchvision/prototype/transforms/_geometry.py [hunk 5]
```diff
@@ -672,3 +672,101 @@ def _get_params(self, sample: Any) -> Dict[str, Any]:
...
+                    F.clamp_bounding_box(inpt[params["is_valid"]], format=inpt.format, image_size=inpt.image_size),
+                )
+
+        if params["needs_pad"]:
+            inpt = F.pad(inpt, params["padding"], fill=self.fill, padding_mode=self.padding_mode)
+
+        return inpt
+
+    def forward(self, *inputs: Any) -> Any:
+        # FIXME: revisit after https://github.com/pytorch/vision/pull/6401#discussion_r948749012 is resolved
```
Comment (pmeier) [inline torchvision/prototype/transforms/_geometry.py:761]: Blocked by #6447.

Comment (datumbox) [reply torchvision/prototype/transforms/_geometry.py]: You should be OK now.
